### PR TITLE
Fix(InputItem): add try catch to setSelection (#3120)

### PR DESCRIPTION
**Pull request by zack24q**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/3124)

### CODE REVIEW

**Comment by doxiaodong** [comment]
从根源上解决error

**Comment by zack24q** [comment]
根本原因是一些老内核下，有一些input type不支持selection api，所以这边catch住之后就做降级处理，不对光标做位置计算，回归老逻辑
兼容性参考链接：https://developer.mozilla.org/zh-CN/docs/Web/API/HTMLInputElement/setSelectionRange#%E6%B5%8F%E8%A7%88%E5%99%A8%E5%85%BC%E5%AE%B9%E6%80%A7

**Comment by doxiaodong** [comment]
https://github.com/ant-design/ant-design-mobile/pull/3124/files#diff-da561ee58a9d0bf6841793d9067364a5R219 
我的意思是这一个行数没发现哪个会抛出 error

**Comment by zack24q** [comment]
相关issue #3120 
对 `selectionStart` 或 `selectionEnd` 赋值可能会抛出error的

File: components/input-item/index.tsx [hunk 1]
```diff
@@ -127,7 +127,11 @@ class InputItem extends React.Component<InputItemProps, any> {
         case 'phone':
         case 'number':
           // controlled input type needs to adjust the position of the caret
-          el.selectionStart = el.selectionEnd = this.calcPos(prePos || 0, preCtrlVal, rawVal, ctrlValue, [' '], /\D/g)
+          try {
+            el.selectionStart = el.selectionEnd = this.calcPos(prePos || 0, preCtrlVal, rawVal, ctrlValue, [' '], /\D/g);
```
Comment (doxiaodong) [inline components/input-item/index.tsx]: 注释一下这里 selectionEnd 会 error ，不然后面的人看不懂

Comment (zack24q) [reply components/input-item/index.tsx]: done

**Pull request merged**